File: Recursion/Lecture 03/CodeStudio/Nth Fibonacci Number.java
import java.util.*;
import java.io.*;
public class Solution {
    // this function fibo() will return us the nth fibonacci number
	public static int fibo(int n){

        // base condition
		if(n==1 || n==2) return 1;

        // recursion call that will return (n-1)th & (n-2)th fibo number which will then be added to give us the nth fibo number
		return fibo(n-1)+fibo(n-2); 
	}

	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int n=sc.nextInt();
		System.out.println(fibo(n));
	}

}

/*
    time complexity : O(2^n)
    space complexity : O(n)
*/

File: Recursion/Lecture 03/CodeStudio/Nth Fibonacci Number.cpp
#include <iostream>
using namespace std;

// this function fibo() will return us the nth fibonacci number
int fibo(int n)
{
    // base condition
    if (n == 1 || n == 2)
        return 1;

    //  recursion call that will give us the (n-1)th fibo number
    int ans1 = fibo(n - 1);

    // recursion call that will give us the (n-2)th fibo number
    int ans2 = fibo(n - 2);

    // will return the addition of (n-1)th & (n-2)th fibo number which is actually our nth fibo number that we need
    return ans1 + ans2;     
}
int main()
{
    // Write your code here.
    int n;
    cin >> n;
    n = fibo(n);
    cout << n;
}

/*
    time complexity : O(2^n)
    space complexity : O(n)
*/
